tests/Unit/DataStructures/Tensor/Test_CreateView.cpp

```diff
@@ -0,0 +1,66 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/CreateView.hpp"
+#include "DataStructures/Tensor/IndexType.hpp"
+#include "DataStructures/Tensor/Metafunctions.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+
+namespace {
+
+SPECTRE_TEST_CASE("Unit.DataStructures.Tensor.CreateView",
+                  "[DataStructures][Unit]") {
+  const size_t SpatialDim = 3;
+  const DataVector used_for_size(5);
+  MAKE_GENERATOR(generator);
+  const auto nn_gen = make_not_null(&generator);
+  std::uniform_real_distribution<> distribution(-1.0, 1.0);
+  const auto nn_dist = make_not_null(&distribution);
+
+  tnsr::a<double, SpatialDim, Frame::Inertial> test_spacetime_vector;
+  const auto scalar_time_component =
+      make_with_random_values<Scalar<double>>(nn_gen, nn_dist, used_for_size);
+  const auto vector_spatial_components =
+      make_with_random_values<tnsr::i<double, SpatialDim, Frame::Inertial>>(
+          nn_gen, nn_dist, used_for_size);
+
+  create_view<SpatialDim, UpLo::Lo, Frame::Inertial>(
```

Comment:
Can the compiler infer the template arguments?  If not, document what they mean.